### PR TITLE
Added delegate method

### PR DESCRIPTION
**Pull request by joaoritter**
Added willChangeState delegate method and updated README

### CODE REVIEW

**Comment by joaoritter** [comment]
I got a line-length violation, but the 10 lines that I added are all the same length as yours...

File: Source/ScrollingNavigationController.swift [hunk 1]
```diff
@@ -8,6 +8,10 @@ import UIKit
      Called when the state of the navigation bar changes
      */
     optional func scrollingNavigationController(controller: ScrollingNavigationController, didChangeState state: NavigationBarState)
+
+    /*******joaoritter*******/
+    optional func scrollingNavigationController(controller: ScrollingNavigationController, willChangeState state: NavigationBarState)
```
Comment (joaoritter) [inline Source/ScrollingNavigationController.swift]: It's the same length as the previous line in master...

Comment (andreamazz) [inline Source/ScrollingNavigationController.swift]: Don't worry about that, but please  remove the ****** comments and use proper comments describing the func as the one above it.

**Comment by andreamazz** [comment]
I manually integrated the changes in 3.0.0 alongside swift 3.

**Pull request closed without merging**